### ISSUE TITLE
Access sound configuration inside controller mapping

### ISSUE THREAD

**Issue by whanake-music** [mixxxdj/mixxx#11550]
### Feature Description

I'm going to use the Denon Prime 4 as an example here, but this applies to multiple controllers as well.

The Prime 4 has a built-in soundcard that Mixxx can connect to in order to make use of the Prime 4's various inputs and outputs.

The Prime 4 also has a master gain knob, microphone inputs with built-in talkback and gain adjustment, and headphone split/mix controls.

These controls all send MIDI that can be mapped to their respective controls in Mixxx via Computer Mode, but they also control the Prime 4's onboard soundcard directly, which results in exponential changes to volume levels when Mixxx is routed through the Prime 4's soundcard.

So my question is: can Mixxx expose its Sound Hardware configuration to mapping files? This way, controllers like the Prime 4 can detect whether Mixxx is using their soundcard directly, and disable the connections between their gain controls and Mixxx as a result. Or, vice versa, if Mixxx is not being routed through their soundcard, then those controls can be connected to Mixxx, allowing for more versatility across different sound configurations.

**Comment by Swiftb0y** [comment]
Unfortunately not a super easy task to implement since we don't know of any cross-platform way to reliably match the USB Midi/HID endpoint with the audio endpoint... (AFAIK)

**Comment by daschuer** [comment]
I can also confirm the issue. A workround would be an entry in the mapping file that holds all info for identification. 

The more general issue is that Mixxx not yet has a real master volume knob. It has a master gain that works in the digital doman also reducing the bitdepth.

Almost all soundcards have a digital controlled analog emplifier for that. I would love to see it used from Mixxx.

Related bugs is https://github.com/mixxxdj/mixxx/issues/7942

**Comment by JoergAtGithub** [comment]
@daschuer This is right for soundcards, but not for controllers and mixers with integrated soundcard. They have usually hardware gain controls for Master + Booth gain, which can't be overwritten by software. You can only read out the position of the knobs in your mapping.
What (all) other DJ software does, is to just visualize the position of these hardware knobs in the GUI and keep the software internal gain at 0dB.

**Comment by daschuer** [comment]
Yes, my controller is also of that type. My point is just that when we consider solutions for hardware volume controls, that we have the whole picture in mind.
